### ISSUE TITLE
Steam Overlay not working

### ISSUE THREAD

**Issue by AngelicJoker** [rebtd7/FF13Fix#51]
Whenever I enable the fix mod in FF XIII, the Steam overlay stops working.  This seems to stop my PS4 controller from working as well.  I am able to use both the controller and overlay if I disable the mod.

**Comment by rebtd7** [comment]
Seems to be caused by e0e756333fd8ebbb3e5925bd1c96b7bc70588db6

@Nucleoprotein Any idea on what may be causing this? I also get no overlay without dxvk

**Comment by AngelicJoker** [comment]
Is dxvk used outside of Linux systems?  I'm running Windows 10 so I don't think I have the option to use it.

**Comment by nav2max** [comment]
> Seems to be caused by [e0e7563](https://github.com/rebtd7/FF13Fix/commit/e0e756333fd8ebbb3e5925bd1c96b7bc70588db6)
> 
> @Nucleoprotein Any idea on what may be causing this? I also get no overlay without dxvk
> 
> Edit: I just saw this:
> [doitsujin/dxvk#1467](https://github.com/doitsujin/dxvk/issues/1467)
> 
> > > This isn't our bug. The overlay on x86 just won't hook loose d3d9.dlls. (it works on x64)
> 
> Perhaps we have to create both `dinput8.dll` and `d3d9.dll`, and tell users to use `dinput8.dll` if they are not using `dxvk`, and tell them to use `d3d9.dll` if they are using `dxvk`?

For me d3d9.dll doesn't work, using Intel i7-9750H processor with an RTX 2070 Max-Q [MSI GS65 Stealth [9S]], but when i revert back to 1.46 13Fix mod, it works perfectly fine. Which means it accepts dinput8.dll but not d3d9.dll

Not sure why it's causing an issue in my Laptop.

**Comment by Nucleoprotein** [comment]
@rebtd7 
It's reference count hell, GameOverlayRenderer is deleting own hooks because object reference is 0 (or too low?)
I have a working fix already but i dont like it because our private ref count does not match in some points D3D9 one... Maybe we should just return D3D9 refcount + 1 always...

**Comment by rebtd7** [comment]
Thanks a lot, @Nucleoprotein!

@AngelicJoker, @nav2max Could you try 1.6.1 and let us know if it is working?

**Comment by Surihix** [comment]
Should the dll files still be renamed similarly like how it was with 1.6.0 version if DXVK is used with this new version ?

**Comment by rebtd7** [comment]
@Surihix Yes, it is the same (FF13Fix is `d3d9.dll` and DXVK (if present) is `dxvk.dll`)

**Comment by AngelicJoker** [comment]
@rebtd7 I just installed the update and now the Steam overlay and PS4 controller are working.  Thanks for the quick response!

**Comment by nav2max** [comment]
> Thanks a lot, @Nucleoprotein!
> 
> @AngelicJoker, @nav2max Could you try 1.6.1 and let us know if it is working?

Sorry it's not even starting the game...not sure what i got wrong. [Currently using Nvidia Profile Inspector graphic settings along with FFXIII HD project & models mod] - All this was working fine when i used 1.46 13Fix file with dinput8.dll
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6177431/FF13Fix.log)

**Comment by Nucleoprotein** [comment]
Looks like object is destroyed (yeah... ref counting) ...
Please check if this works
[d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178137/d3d9.zip)

**Comment by nav2max** [comment]
> Looks like object is destroyed (yeah... ref counting) ...
> Please check if this works
> [d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178152/d3d9.zip)

Nope no luck, it didn't even open in default settings, but when i changed the settings by enabling V-sync, limiting to 60 FPS and turning borderless ON, it almost started...
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6178178/FF13Fix.log)

**Comment by Nucleoprotein** [comment]
This?
[d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178195/d3d9.zip)

**Comment by nav2max** [comment]
> This?
> [d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178195/d3d9.zip)

Nope, checked now, no luck
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6178202/FF13Fix.log)

**Comment by Nucleoprotein** [comment]
disable DiscardUIVertexBuffer

**Comment by nav2max** [comment]
> disable DiscardUIVertexBuffer

It's currently set to default "false"

**Comment by Nucleoprotein** [comment]
You have too many options changed to unsafe values, create new clean config file.

**Comment by nav2max** [comment]
> You have too many options changed to unsafe values, create new clean config file.
> Patching thread is not even starting ...

This time i placed a new config file, still no luck, DiscardUIVertexBuffer is on default "false"
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6178223/FF13Fix.log)

**Comment by Nucleoprotein** [comment]
It must be wrong ref count...
[Uploading d3d9.zip…]()

**Comment by nav2max** [comment]
No luck...
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6178329/FF13Fix.log)

**Comment by Nucleoprotein** [comment]
This one is broken i.e. not call Release() at all, if this works... i'm confused
[d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178361/d3d9.zip)

**Comment by nav2max** [comment]
> This one is broken i.e. not call Release() at all, if this works... i'm confused
> [d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178361/d3d9.zip)

Nope...just my bad luck, not working..
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6178414/FF13Fix.log)

**Comment by nav2max** [comment]
> This one is broken i.e. not call Release() at all, if this works... i'm confused
> [d3d9.zip](https://github.com/rebtd7/FF13Fix/files/6178361/d3d9.zip)

I fixed it.... !!!! I disabled Windows 7 compatibility mode and unticked "Disable Full screen Optimization", it works now...no issue
[FF13Fix.log](https://github.com/rebtd7/FF13Fix/files/6178432/FF13Fix.log)

**Comment by Nucleoprotein** [comment]
It was W7 compatibility mode, original one should work too

**Comment by l3illyl3ob** [comment]
On 1.6.1 and 1.6.2, the overlay and controller input works for me without dxvk, but now it doesn't work for me with dxvk. Does the 1.6.1 fix not apply to dxvk users? Is there a workaround?
